store/builder.go

```diff
@@ -271,3 +241,95 @@ func (d *Builder) BuildManager(
 	)
 	return NewManager(eigenDAV1Store, eigenDAV2Store, s3Store, d.log, secondary, d.v2ClientCfg.Enabled)
 }
+
+func (d *Builder) buildEthClient() (common_da.EthClient, error) {
+	gethCfg := geth.EthClientConfig{
+		RPCURLs: []string{d.v1EdaClientCfg.EthRpcUrl},
+	}
+
+	ethClient, err := geth.NewClient(gethCfg, geth_common.Address{}, 0, d.log)
+	if err != nil {
+		return nil, fmt.Errorf("create geth client: %w", err)
+	}
+
+	return ethClient, nil
+}
+
+func (d *Builder) buildRelayPayloadRetriever(
+	ethClient common_da.EthClient,
+	maxBlobSizeBytes uint,
+	g1Srs []bn254.G1Affine,
+) (*clients_v2.RelayPayloadRetriever, error) {
+	relayClient, err := d.buildRelayClient(ethClient, maxBlobSizeBytes)
+	if err != nil {
+		return nil, fmt.Errorf("build relay client: %w", err)
+	}
+
+	relayPayloadRetriever, err := clients_v2.NewRelayPayloadRetriever(
+		d.log,
+		//nolint:gosec // disable G404: this doesn't need to be cryptographically secure
+		rand.New(rand.NewSource(time.Now().UnixNano())),
+		d.v2ClientCfg.RelayPayloadRetrieverCfg,
+		relayClient,
+		g1Srs)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create relay payload retriever: %w", err)
```

Comment:
nit and probably my doing - error msg misaligns with existing wrapping patterns for construction err messages

Reply:
[8738083](https://github.com/Layr-Labs/eigenda-proxy/pull/317/commits/8738083d3ac2ba4880e4f14c34abf2446df04250)